lib/galaxy/managers/workflows.py

```diff
@@ -720,6 +720,7 @@ def _workflow_from_raw_description(
                     source_metadata["trs_tool_id"] = workflow_state_resolution_options.trs_tool_id
                     source_metadata["trs_version_id"] = workflow_state_resolution_options.trs_version_id
                     source_metadata["trs_server"] = workflow_state_resolution_options.trs_server
+                    source_metadata["trs_url"] = workflow_state_resolution_options.trs_url
```

Comment:
I wasn't sure what this `source_metadata` is used for, and if there are any consequences to a workflow not having a `trs_server` defined if it was imported directly from a URL.

Reply:
You're capturing all the relevant information and the tests below are really nice. I think the goal was to be able to upgrade the workflow after upgrades and to provide linkage information in the UI - I think what you're capturing here is sufficient to do either.